client/client.go

```diff
@@ -1,19 +1,163 @@
-package client
+package main
 
 import (
 	"context"
+	"crypto/rand"
+	"flag"
+	"fmt"
+
+	log "github.com/golang/glog"
 
 	"github.com/openconfig/bootz/proto/bootz"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
 )
 
-type Client struct {
-	bootz.BootstrapClient
-}
+// Represents a 128 bit nonce.
+const nonceLength = 16
 
-func (c *Client) GetBootstrapData(ctx context.Context, req *bootz.GetBootstrapDataRequest, opts ...grpc.CallOption) (*bootz.GetBootstrapDataResponse, error) {
+var (
+	bootMode = flag.String("boot_mode", "SecureOnly", "The BootMode the device can start in.")
+	port     = flag.String("port", "", "The port to listen to on localhost for the bootz server.")
+)
 
+// generateNonce() generates a fixed-length nonce.
+func generateNonce() (string, error) {
+	b := make([]byte, nonceLength)
+	_, err := rand.Read(b)
+	if err != nil {
+		return "", err
+	}
+	return string(b), nil
 }
-func (c *Client) ReportStatus(ctx context.Context, req *bootz.ReportStatusRequest, opts ...grpc.CallOption) (*bootz.EmptyResponse, error) {
 
+func main() {
+	ctx := context.Background()
+	flag.Parse()
+
+	secureOnly := *bootMode == "SecureOnly"
+
+	// Construct the fake device.
+	// TODO: Allow these values to be set e.g. via a flag.
+	chassis := bootz.ChassisDescriptor{
+		Manufacturer: "Cisco",
+		SerialNumber: "123",
+		ControlCards: []*bootz.ControlCard{
+			{
+				SerialNumber: "123A",
+				Slot:         1,
+				PartNumber:   "123A",
+			},
+			{
+				SerialNumber: "123B",
+				Slot:         2,
+				PartNumber:   "123B",
+			},
```

Comment:
We should consider moving these to a single .conf file in future CLs. 

One potential benefit of doing this would be to only run server or client to test the other side (which is not mocked), that may require some more realistic data.

Reply:
same goes for url and port below, those could be taken from either a single conf file or from command line or both